02-starting-with-data.Rmd

```diff
@@ -111,10 +111,132 @@ Based on the output of `str(surveys)`, can you answer the following questions?
+surveys[1, 6]   # first element in the 6th column (as a vector)
+surveys[1:3, 7] # first three elements in the 7th column (as a vector)
+surveys[3, ]    # the 3rd element for all columns (as a data.frame)
+head_surveys <- surveys[1:6, ] # equivalent to head(surveys)
+```
+
+`:` is a special function that creates numeric vectors of integers in increasing
+or decreasing order, test `1:10` and `10:1` for instance.
+
+You can also exclude certain parts of a data frame using the "`-`" sign:
+
+```{r, purl=FALSE}
+surveys[,-1]   #The whole data frame, except the first column
+surveys[-c(7:34786),] #equivalent to head(surveys)
+```
+
+As well as using numeric values to subset a `data.frame` (or `matrix`), columns
+can be called by name, using one of the three following notations:
+
+```{r, eval = FALSE, purl=FALSE}
+surveys["species_id"]       # Result is a data.frame
+surveys[, "species_id"]     # Result is a vector
+surveys[["species_id"]]     # Result is a vector
+surveys$species_id          # Result is a vector
+```
+
+For our purposes, the last three notations are equivalent. However, the last one
+with the `$` does partial matching on the name. So you could also select the
+column `"day"` by typing `surveys$d`. It's a shortcut, as with all shortcuts,
+they can have dangerous consequences, and are best avoided. Besides, with
+auto-completion in RStudio, you rarely have to type more than a few characters
+to get the full and correct column name.
+
+> ### Challenge
+>
+> 1. Create a `data.frame` containing only the observations from row 1999 of the
+>    `surveys` dataset.
+>
+> 2. Notice how `nrow()` gave you the number of rows in a `data.frame`? Use
+>   `nrow()` instead of a row number to make a `data.frame` with observations from
+>   only the last row of the `surveys` dataset.
+>
+> 3. Use `nrow()` in conjunction with the `:` function to create a new
+>    `data.frame` called `surveys_last_100` that includes the 100 rows of the
+>    survey data frame.
+>
+> 4. Combine `nrow()` with the `-` notation above to reproduce the behavior of
+>   `head(surveys)` excluding the 7th through final row of the `surveys` dataset.
+
+```{r, echo=FALSE, purl=TRUE}
+### Challenges:
+###
+###  1. Create a `data.frame` (`surveys_1999`) containing only the
+###    observations from row 1999 of the `surveys` dataset.
+###
+### 2. Notice how `nrow()` gave you the number of rows in a
+###   `data.frame`? Use `nrow()` instead of a row number to make a
+###   `data.frame` (`surveys_last`) with observations from only the
+###   last row of the `surveys` dataset.
+###
+### 3. Use `nrow()` in conjunction with the `:` function to create a new
+###    `data.frame` called `surveys_last_100` that includes the 100 rows of the
+###    survey data frame.
+###
+### 4. Combine `nrow()` with the `-` notation above to reproduce the behavior of
+###   `head(surveys)` excluding the 7th through final row of the `surveys` dataset.
+###
+```
```

Comment:
This is a repeat of the above challenges so that it ends up in the Rmd code document that students have. So, this would just have to be updated to reflect any challenge changes.